posts/2023-06-14-running-spring-boot-3.adoc

```diff
@@ -0,0 +1,163 @@
+
+To run this application on Open Liberty, you can add the link:https://github.com/OpenLiberty/ci.maven[Liberty Maven plugin] to the existing Petclinic `pom.xml` file. You can also configure a package that runs the required `servlet-6.0` feature and other helpful features by default.
+
+1. Clone the sample repository to your machine.
++
+[source,sh]
+----
+git clone https://github.com/spring-projects/spring-petclinic.git 
+cd spring-petclinic
+----
+
+2. Edit the Petclinic `pom.xml` file to build the sample application from a WAR file by using the Liberty Maven plugin.
++
+First, change the name of the sample from `petclinic` to `liberty-spring-petclinic` and add the `war` packaging type:
++
+[source,xml]
+----
+<name>liberty-spring-petclinic</name>
+<packaging>war</packaging>
+----
++
+Next, specify a WAR file name under the `build` section:
++
+[source,xml]
+----
+<build>
+    <finalName>liberty-${project.artifactId}</finalName>
+----
++
+And finally, under the `<build> <plugins>` section, add the Liberty Maven plugin:
++
+[source,xml]
+----
+<plugin>
+  <groupId>io.openliberty.tools</groupId>
+  <artifactId>liberty-maven-plugin</artifactId>
+  <version>3.8.1</version>
+  <configuration>
+      <runtimeArtifact>
+          <groupId>io.openliberty</groupId>
+          <artifactId>openliberty-webProfile10</artifactId>
+          <version>23.0.0.5</version>
+          <type>zip</type>
+      </runtimeArtifact>
+  </configuration>
+</plugin>
+----
++
+This example specifies a recent version of the `webProfile10` package, which includes the required `servlet-6.0` feature.
+
+3. Add the Servlet Initializer class to the Petclinic application under `/src/main/java` 
++
+This class extends the `SpringBootServletInitializer` class and points to the sample Petclinic application.
++
+[source,java]
+----
+PetClinicServletInitializer.java
+
+    package org.springframework.samples.petclinic;
+
+    import org.springframework.boot.builder.SpringApplicationBuilder;
+    import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
+
+    public class PetClinicServletInitializer extends SpringBootServletInitializer {
+
+        @Override
+        protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+            return application.sources(PetClinicApplication.class);
+        }
+
+    }
+----
+
+Now we are ready to compile and run!
+
+== Compile and run the Petclinic sample application
+
+1. Run maven clean to clear out any existing classes from a previous compile
++
+[source,sh]
+----
+$ ./mvnw clean
+----
+
+2. Compile the project to ensure the initializer class looks good and that you were able to download all the necessary dependencies.
++
+[source,sh]
+----
+$ ./mvnw compile
+----
+
+3. Use the Liberty Maven plugin to define a "loose application" based on the sample application, which automatically deploys all the source and dependencies to Liberty. This command also downloads and starts the server.
```

Comment:
Changing this line to say:
3. The liberty plugin command downloads the liberty runtime, finds the packaged war file (liberty-spring-petclinic.war) and deploys to the dropins directory.  and starts the server.